connection.go

```diff
@@ -549,22 +567,22 @@ func (c *Connection) Insecure() bool {
 
 // AccountsMgmt returns the client for the accounts management service.
 func (c *Connection) AccountsMgmt() *accountsmgmt.Client {
-	return accountsmgmt.NewClient(c, "/api/accounts_mgmt", "/api/accounts_mgmt")
+	return accountsmgmt.NewClient(c.transport, "/api/accounts_mgmt", "/api/accounts_mgmt")
 }
 
 // ClustersMgmt returns the client for the clusters management service.
 func (c *Connection) ClustersMgmt() *clustersmgmt.Client {
-	return clustersmgmt.NewClient(c, "/api/clusters_mgmt", "/api/clusters_mgmt")
+	return clustersmgmt.NewClient(c.transport, "/api/clusters_mgmt", "/api/clusters_mgmt")
```

Comment:
If you do this then the transport wrapper will apply only to the type safe interface of the SDK. It will not apply to uses like this:

```go
response, err := connection.Get("/api/clusters_mgmt/v1/clusters").
       Send()
if err != nil {
        ...
}
```

It won't apply to the requests used to get tokens either.

Reply:
@jhernand what is your suggestion?

Reply:
Something like this:

```diff
@@ -453,16 +454,22 @@ func (b *ConnectionBuilder) BuildContext(ctx context.Context) (connection *Conne
                agent = DefaultAgent
        }
 
-       // Create the HTTP client:
+       // Create the transport:
        // #nosec 402
-       client := &http.Client{
-               Transport: &http.Transport{
-                       TLSClientConfig: &tls.Config{
-                               InsecureSkipVerify: b.insecure,
-                               RootCAs:            b.trustedCAs,
-                       },
+       var transport http.RoundTripper = &http.Transport{
+               TLSClientConfig: &tls.Config{
+                       InsecureSkipVerify: b.insecure,
+                       RootCAs:            b.trustedCAs,
                },
        }
+       if b.wrapper != nil {
+               transport = b.wrapper(transport)
+       }
+
+       // Create the HTTP client:
+       client := &http.Client{
+               Transport: transport,
+       }
 
        // Allocate and populate the connection object:
        connection = &Connection{
```

That will configure the HTTP client that is used for all HTTP communications (raw, type safe and tokens) to use the wrapped transport.

Reply:
Fixed. PTAL